gdmix-trainer/setup.py

```diff
@@ -30,22 +30,23 @@
     packages=find_namespace_packages(where='src'),
     include_package_data=True,
     install_requires=[
+        "numpy==1.19.5",
         "absl-py==0.10",
         "decorator==4.4.2",
-        "detext-nodep==3.0.0",
-        "fastavro==1.4.7",
-        "google-auth==1.25.0",
-        "google-cloud-bigquery==2.18.0",
+        "detext-nodep==3.2.0",
+        "gin-config==0.3.0",
+        "fastavro==0.21.22",
         "grpcio==1.32.0",
-        "numpy==1.19.5",
         "protobuf==3.19",
         "psutil==5.7.0",
         "scikit-learn==1.0",
         "setuptools>=41.0.0",
         "six==1.15.0",
-        "smart-arg==0.4",
+        "smart-arg==0.3.4",
```

Comment:
why downgrade smart-arg?

Reply:
There was an issue from argument serialization in 0.4 so I downgraded it, since we don't use its argument serialization anymore, I'll use 0.4 in the next update.

Reply:
just curious, what issue? 0.4 is a very old version and does the latest version have the same issue? Thanks.

Reply:
@jakiejj It's related to smartArgs argument serialization but I don't remember the detail. I gave up smartArgs argument serialization also because it doesn't support serializing arguments whose default value is None, which occurs to  DeText's new version.

Reply:
Thanks. smartArg does support None default. It has to be an "Optional" though.

Reply:
Yeah, I think it's DeText's "bug"

Reply:
add back smart-arg==0.4